aws/data_source_aws_kms_alias.go

```diff
@@ -64,25 +64,27 @@ func dataSourceAwsKmsAliasRead(d *schema.ResourceData, meta interface{}) error {
 	d.SetId(time.Now().UTC().String())
 	d.Set("arn", alias.AliasArn)
 
-	// Some aliases do not return TargetKeyId (e.g. aliases for AWS services or
-	// aliases not associated with a Customer Managed Key (CMK))
+	// ListAliases can return an alias for an AWS service key (e.g.
+	// alias/aws/rds) without a TargetKeyId if the alias has not yet been
+	// used for the first time. In that situation, calling DescribeKey will
+	// associate an actual key with the alias, and the next call to
+	// ListAliases will have a TargetKeyId for the alias.
+	//
+	// For a simpler codepath, we always call DescribeKey with the alias
+	// name to get the target key's ARN and Id direct from AWS.
+	//
 	// https://docs.aws.amazon.com/kms/latest/APIReference/API_ListAliases.html
-	if alias.TargetKeyId != nil {
-		aliasARN, err := arn.Parse(*alias.AliasArn)
-		if err != nil {
-			return err
-		}
-		targetKeyARN := arn.ARN{
-			Partition: aliasARN.Partition,
-			Service:   aliasARN.Service,
-			Region:    aliasARN.Region,
-			AccountID: aliasARN.AccountID,
-			Resource:  fmt.Sprintf("key/%s", *alias.TargetKeyId),
-		}
-		d.Set("target_key_arn", targetKeyARN.String())
 
-		d.Set("target_key_id", alias.TargetKeyId)
+	req := &kms.DescribeKeyInput{
+		KeyId: aws.String(target.(string)),
 	}
+	resp, err := conn.DescribeKey(req)
+	if err != nil {
+		return errwrap.Wrapf("Error calling KMS DescribeKey: {{err}}", err)
```

Comment:
FYI I know there are a bunch of places in the repository currently using this, but we don't need the context wrapping of `errwrap` when we're just returning an error message. `fmt.Errorf` is just fine 😄

Reply:
We'll do it later, no sense holding this up. 👍